test/e2e/singlecluster/visibility_test.go

```diff
@@ -500,16 +500,20 @@ var _ = ginkgo.Describe("Kueue visibility server", func() {
 
 		ginkgo.It("Should return an appropriate error", func() {
 			ginkgo.By("Returning a ResourceNotFound error for a nonexistent ClusterQueue", func() {
-				_, err := impersonatedVisibilityClient.ClusterQueues().GetPendingWorkloadsSummary(ctx, "non-existent", metav1.GetOptions{})
-				statusErr, ok := err.(*errors.StatusError)
-				gomega.Expect(ok).To(gomega.BeTrue())
-				gomega.Expect(statusErr.ErrStatus.Reason).To(gomega.Equal(metav1.StatusReasonNotFound))
+				gomega.Eventually(func(g gomega.Gomega) {
```

Comment:
Can you explain why wrapping the code in Eventually helps here. If the CQ is non-existent then we should expect `StatusReasonNotFound` consistently.

Reply:
Yes, you right, but if `ClusterRoleBinding` not applied we will get `StatusReasonForbidden`. IIRC it can't apply immediately. Correct me if I'm wrong.

Reply:
I don't know this bit, please investigate if you can, but if this is the case, I think the preferred solution is to wait for that to happen inside `BeforeEach` rather than inside individual `It`s.

Reply:
I'm ok with this approach.

I think the delay makes sense. IIUC, it is Kueue that is verifying the RBAC, and the role objects can be delayed due to the nature of the informers.

@PBundyra does this match your knowledge of the apiserver extensions?

Reply:
I think we can do the same - use eventually inside BeforeEach. Then the active waiting will not need to be spread across `It`s, and easier to replace if we even find out another approach to wait.

Reply:
>> I think the delay makes sense. IIUC, it is Kueue that is verifying the RBAC, and the role objects can be delayed due to the nature of the informers.

IIRC it's apiserver's responsibility to authorize (based on RBAC) requests from users. Refering to [Authentication Flow documentation](https://kubernetes.io/docs/tasks/extend-kubernetes/configure-aggregation-layer/#authentication-flow)

"_Kubernetes apiserver: authenticate the requesting user and authorize their rights to the requested API path._"

Even at the stage of authorizing request by extension apiserver (Kueue in this case) is still asks apiserver:

"_The extension apiserver now can validate that the user/group retrieved from the headers are authorized to execute the given request. It does so by sending a standard [SubjectAccessReview](https://kubernetes.io/docs/reference/access-authn-authz/authorization/) request to the Kubernetes apiserver._"